src/test/java/banking/backend/MoneyTest.java

```diff
@@ -110,42 +106,56 @@ void equalsAndConstructor() {
-                () -> assertThrows(NumberFormatException.class, () -> new Money("100.111€")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("10,0€")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("10.€")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("100.100,0€")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("100.100,01€")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("10")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("0")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("0.00")),
-                () -> assertThrows(NumberFormatException.class, () -> new Money("10,0€")),
-                () -> assertThrows(IllegalArgumentException.class, () -> new Money(10, -1)),
-                () -> assertThrows(IllegalArgumentException.class, () -> new Money(10, 100))
-        );
+        assertFalse(oneInt.equals(null));
+        assertFalse(oneInt.equals(new Object()));
+        assertFalse(oneInt.equals("1"));
+        assertFalse(oneInt.equals(1));
+        assertNotEquals(oneInt, twoInt);
+        assertEquals(oneCentInt, oneCentDouble);
+        assertEquals(oneCentInt, oneCentString);
+        assertEquals(oneCentDouble, oneCentString);
+        assertNotEquals(oneCentInt, oneIntInt);
+        assertNotEquals(oneInt, oneCentInt);
+        assertEquals(new Money("0.00€"), new Money("0€"));
+        assertEquals(new Money("-0.00€"), new Money("0€"));
+        assertThrows(NumberFormatException.class, () -> new Money(""));
+        assertThrows(NumberFormatException.class, () -> new Money("100.111€"));
+        assertThrows(NumberFormatException.class, () -> new Money("10,0€"));
+        assertThrows(NumberFormatException.class, () -> new Money("10.€"));
+        assertThrows(NumberFormatException.class, () -> new Money("100.100,0€"));
+        assertThrows(NumberFormatException.class, () -> new Money("100.100,01€"));
+        assertThrows(NumberFormatException.class, () -> new Money("10"));
+        assertThrows(NumberFormatException.class, () -> new Money("0"));
+        assertThrows(NumberFormatException.class, () -> new Money("0.00"));
+        assertThrows(NumberFormatException.class, () -> new Money("10,0€"));
+        assertThrows(IllegalArgumentException.class, () -> new Money(10, -1));
+        assertThrows(IllegalArgumentException.class, () -> new Money(-10, 1));
+        assertThrows(IllegalArgumentException.class, () -> new Money(10, 100));
     }
 
     /**
      * Tests several adds of Percentages.
      */
     @Test
     void applyPercentage() {
-        throw new NotYetImplementedException();
-    }
+        assertEquals(
+                new Money(1, 5),
+                new Money(1).applyPercentage(new Percentage(105)));
```

Comment:
Test tests wrong cases please change this to
`new Money(1, 5), `
`new Money(1).applyPercentage(new Percentage(5)));`